lib/r10k/module/base.rb

```diff
@@ -60,8 +60,14 @@ def initialize(title, dirname, args, environment=nil)
     @environment = environment
     @overrides = args.delete(:overrides) || {}
     @spec_deletable = true
-    @exclude_spec = args.delete(:exclude_spec)
-    @exclude_spec = @overrides[:modules].delete(:exclude_spec) if @overrides.dig(:modules, :exclude_spec)
+    @exclude_spec = case
+                    when @overrides.dig(:modules, :exclude_spec) == false
+                      false
+                    when @args.delete(:exclude_spec) == false
+                      false
+                    else
+                      true
+                    end
```

Comment:
I think in Molly's suggestion ending with `|| true` will always have the value set to `true`, eg you'll never be able to set the value to `false`.

I think you want to do the digging and then check for `nil?` ie, get whatever value it's been set to, and then, if it hasn't been set it will be `nil`, and in that case you want `true`.

Reply:
oooh, I see what you mean, my bad.

Reply:
I was thinking that these values would be unset (nil) if a user had not passed them, however, I realize now that at least one of them will be true by default (kinda the point of this pr). So I think we do need these explicit false checks.

Reply:
The benefit of deleting the value from the map is that we ensure we aren't inadvertently adding logic around that value erroneously after the base initialization. I moved that to execute outside the case so we ensure it always happens.

Reply:
Circling back to this; there is never a time when `exclude_spec` is set in the args; that was an error in logic from the original PR to add this setting, so I have removed that from this PR in the upgrade to 4.0.